Index: Lib/fontTools/ttLib/tables/ttProgram.py
===================================================================
--- Lib/fontTools/ttLib/tables/ttProgram.py	(revision 562)
+++ Lib/fontTools/ttLib/tables/ttProgram.py	(revision 563)
@@ -394,7 +394,7 @@
 	
 	p = Program()
 	p.fromBytecode(bc)
-	as = p.getAssembly()
-	p.fromAssembly(as)
+	asm = p.getAssembly()
+	p.fromAssembly(asm)
 	print bc == p.getBytecode()
 
